$OpenBSD: patch-osx_ml,v 1.1 2005/05/30 19:27:45 sturm Exp $
Post-release developer patch.
Fix the Mac OS X client problem.
See http://lists.seas.upenn.edu/pipermail/unison-hackers/2005-May/000076.html
--- osx.ml.orig	Sat Mar 12 11:17:59 2005
+++ osx.ml	Mon May 23 13:50:28 2005
@@ -386,9 +386,10 @@ let ressDummy = NoRess
 let openRessIn fspath path =
   Util.convertUnixErrorsToTransient "reading ressource fork" (fun () ->
     try
-      open_in_gen
-        [Open_rdonly; Open_binary] 0o444
-        (Fspath.concatToString fspath (ressPath path))
+      Unix.in_channel_of_descr
+        (Unix.openfile
+           (Fspath.concatToString fspath (ressPath path))
+           [Unix.O_RDONLY] 0o444)
     with Unix.Unix_error (Unix.ENOTDIR, _, _) ->
       let (doublePath, inch, entries) = openDouble fspath path in
       try
